pkg/resource/stack/secrets.go

```diff
@@ -86,171 +89,507 @@ func (s NamedStackSecretsProvider) OfType(ty string, state json.RawMessage) (sec
 		return nil, fmt.Errorf("constructing secrets manager of type %q: %w", ty, err)
 	}
 
-	return NewCachingSecretsManager(sm), nil
+	return NewBatchingCachingSecretsManager(sm), nil
 }
 
-type cacheEntry struct {
-	plaintext  string
-	ciphertext string
+// BatchingSecretsManager is a secrets.Manager that supports batch encryption and decryption operations.
+type BatchingSecretsManager interface {
+	secrets.Manager
+	// BeginBatchEncryption returns a new BatchEncrypter and CompleteCrypterBatch function.
+	// The BatchEncrypter allows encryption operations to be enqueued then processed in a batch request to avoid
+	// round-trips for when the underlying encrypter requires a network call. Encryptions to a secret are enqueued and
```

Comment:
```suggestion
	// round-trips for when doing so with the underlying encrypter incurs overhead. Encryptions to a secret are enqueued and
```

Feels like network stuff is implementation-specific -- really we are just offering an interface for batching operations? Happy either way.